carapace-server/src/main/java/org/carapaceproxy/server/certificates/Route53Client.java

```diff
@@ -0,0 +1,180 @@
+import software.amazon.awssdk.services.route53.model.Change;
+import software.amazon.awssdk.services.route53.model.ChangeAction;
+import software.amazon.awssdk.services.route53.model.ChangeBatch;
+import software.amazon.awssdk.services.route53.model.ChangeResourceRecordSetsRequest;
+import software.amazon.awssdk.services.route53.model.HostedZone;
+import software.amazon.awssdk.services.route53.model.ListHostedZonesByNameRequest;
+import software.amazon.awssdk.services.route53.model.ListHostedZonesByNameResponse;
+import software.amazon.awssdk.services.route53.model.RRType;
+import software.amazon.awssdk.services.route53.model.ResourceRecord;
+import software.amazon.awssdk.services.route53.model.ResourceRecordSet;
+import software.amazon.awssdk.services.route53.model.Route53Response;
+import software.amazon.awssdk.services.route53.model.TestDnsAnswerRequest;
+import software.amazon.awssdk.services.route53.model.TestDnsAnswerResponse;
+
+/**
+ * Client for AWS Route53.
+ *
+ * @author paolo.venturi
+ */
+public class Route53Client {
+
+    public enum DNSChallengeAction {
+        UPSERT,
+        DELETE,
+        CHECK
+    }
+
+    private static final String DNS_CHALLENGE_PREFIX = "_acme-challenge.";
+    private static final String HOSTEDZONE_ID_PREFIX = "/hostedzone/";
+
+    private final Route53AsyncClient client;
+
+    public Route53Client(String awsAccessKey, String awsSecretKey) {
+
+        AwsBasicCredentials awsCreds = AwsBasicCredentials.create(awsAccessKey, awsSecretKey);
+
+        client = Route53AsyncClient.builder()
+                .region(Region.AWS_GLOBAL)
+                .credentialsProvider(StaticCredentialsProvider.create(awsCreds))
+                .build();
+    }
+
+    public void close() {
+        client.close();
+    }
+
+    public void createDNSChallengeForDomain(String domain, String digest, Consumer<Route53Response> onRequestComplete, Consumer<Throwable> onRequestFail) {
+        performActionOnDNSChallengeForDomain(domain, digest, UPSERT, onRequestComplete, onRequestFail);
+    }
+
+    public void deleteDNSChallengeForDomain(String domain, String digest, Consumer<Route53Response> onRequestComplete, Consumer<Throwable> onRequestFail) {
+        performActionOnDNSChallengeForDomain(domain, digest, DELETE, onRequestComplete, onRequestFail);
+    }
+
+    public void isDNSChallengeForDomainAvailable(String domain, String digest, Consumer<Boolean> onRequestComplete, Consumer<Throwable> onRequestFail) {
+        Consumer<Route53Response> check = res -> {
+            TestDnsAnswerResponse r = (TestDnsAnswerResponse) res;
+            onRequestComplete.accept(r.recordData().size() == 1 && r.recordData().get(0).equals("\"" + digest + "\""));
+        };
+        performActionOnDNSChallengeForDomain(domain, null, CHECK, check, onRequestFail);
+    }
+
+    private void performActionOnDNSChallengeForDomain(String domain, String digest, DNSChallengeAction action, Consumer<Route53Response> onRequestComplete,
```

Comment:
if you want this API to be async you have to return a CompletableFuture
having two callbacks is not straightforward for the caller.
alternatively you can have just one callback
interface DNAChallengeCompletionCallback {
        void  onChallengeCompleted(Throwable error);
}